### PR TITLE
Inject data documentation revised

### PR DESCRIPTION
**Pull request by Mihael03**

### CODE REVIEW

File: docs/Actions/inject-data.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+## `Typical Use Cases`
+
+- Generate a new token
+- Concatenate 2 tokens
+- Overwrite a token
+- Initialize variables with empty strings
+
+## `Don't use it to`
+
+- Execute a Variables token - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:21]: What is a "variable token"?

File: docs/Actions/inject-data.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
+---
+id: inject-data
+title: Inject Data
+sidebar_label: Inject Data
+---
+
+> Audience: [`Business Users`](/docs/audience#business-users)<br/>
+> Skill Prerequisites: `None`
+
+An action that lets you add data as tokens to the current execution context. Inject data can be used to load data that other actions other actions down the stack need.
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:10]:
```suggestion
An action that lets you create new tokens into the current execution context. The value of each token is usually a computation of other tokens from the context. Once created, the new tokens can be accessed by any subsequent action.
```

File: docs/Actions/inject-data.md (hunk 1 above)
Comment (bogdan-litescu) [reply docs/Actions/inject-data.md:21]: I suggest something like "Execute tokens that don't return any output" if that's the idea. If not, please make it clearer.

File: docs/Actions/inject-data.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+- Generate a new token
+- Concatenate 2 tokens
+- Overwrite a token
+- Initialize variables with empty strings
+
+## `Don't use it to`
+
+- Execute a Variables token - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Apply tokens inside a text from different sources - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Create a token that is used in another execution context - store the data in a [hidden field](/docs/Actions/hidden-field.md), cookies or server session instead
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:23]: I would suggest mentioning also storing data in Entities or User Profile.

File: docs/Actions/inject-data.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+- Generate a new token
+- Concatenate 2 tokens
+- Overwrite a token
+- Initialize variables with empty strings
+
+## `Don't use it to`
+
+- Execute a Variables token - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Apply tokens inside a text from different sources - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:22]: I don't understand this one as well.

File: docs/Actions/inject-data.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+- Create a token that is used in another execution context - store the data in a [hidden field](/docs/Actions/hidden-field.md), cookies or server session instead
+
+## `Related Actions`
+
+| Action Name | Description|
+|-------------|------------|
+| [Run SQL Query](/docs/Actions/run-sql-query.md) | Executes an SQL statement and captures the output, so that it can be used in the form using Inject Data.|
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:29]: Why do we mention "form" here? Is it specific to a form?
I still don't see a very strong relationship between these 2 actions. Perhaps if you reprhase it to say something like creating new tokens derived from data returned by the SQL query.

What about Execute Actions on Entity List? I remember we sometimes use inject data to create a token that gets appended with some text by each iteration.

File: docs/Actions/inject-data.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+| Parameter     | Description                           | Supports Variables |
+|---------------|---------------------------------------|--------------------|
+| Name/Value | The first parameter refers to the name of the new token that will be created. The second one refers to the value assigned. The action uses a key-value (name-value) concept. | Yes |
+
+## `Output Parameters Reference`
+
+| Parameter | Description |
+|-----------|-------------|
+| Name | The Name parameter is stored as a token for further use anywhere in the form. |
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:41]: I don't see an Output Parameter named "Name" in the action. If you refer to the Name/Value collection, I suggest that we move the entire parameter under Output Parameters section and we don't cover it in Input at all.

File: docs/Actions/inject-data.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+## `Examples`
+
+### `1. Inject data using Run SQL Query`
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:45]: I don't see the value of this example. Why wouldn't I use Inject_Username directly in the SQL action and avoid having the Inject data? Tell me about real-life examples that would help me.

File: docs/Actions/inject-data.md [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+            }
+        ]
+    }
+}
+```
+
+### `2. Generate a new token`
+
+​
+The action below generates a new token named "ClientId" with the assigned value "ID15367" that can be used in the form. <br></br> [Import it](/docs/Actions/import-actions.md) into your application to see it in action.
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:118]: Will this example work only inside the form?
Also, I'm missing the value. When would I need to do that?
Perhaps, a better example for basic usage would be to inject a token that sets something to Debug=true. Then, add another action, such as Debug Logging, that uses the debug token in a condition.

File: docs/Actions/inject-data.md [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+            }
+        ]
+    }
+}
+```
+
+### `4. Overwrite a token`
+
+​
+The action below creates a token with initial value "1" and overwrites it. The injected token will have value "2", which is the new one. <br></br> [Import it](/docs/Actions/import-actions.md) into your application to see it in action.
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:215]: Let's find a real-life example when we'd do this.

File: docs/Actions/inject-data.md [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+| Parameter | Description |
+|-----------|-------------|
+| Name | The Name parameter is stored as a token for further use anywhere in the form. |
+
+## `Examples`
+
+### `1. Inject data using Run SQL Query`
+
+​
+The action below selects first username from users database using Run SQL Query and sends it into the form. <br></br>[Import it](/docs/Actions/import-actions.md) into your application to see it in action.
```
Comment (bogdan-litescu) [inline docs/Actions/inject-data.md:48]: Don't mention "form". This action works in form, grid, API endpoint, scheduler, infobox and workflows.

File: docs/Actions/inject-data.md (hunk 1 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:21]: You're right, it can be used in this circumstance. I deleted it.

File: docs/Actions/inject-data.md (hunk 4 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:22]: I deleted it.

File: docs/Actions/inject-data.md (hunk 3 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:23]: Why? I don't see it as a valid case, or I don't understand what you are referring to

File: docs/Actions/inject-data.md (hunk 5 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:29]: The last suggestion is a "hack" and we suggest using MyTokens instead, usually.

File: docs/Actions/inject-data.md (hunk 6 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:41]: I saw on Radu's commit another structure(that seemed good), so i used it here. I propose we use the same structure everywhere we have a key-value pair.

File: docs/Actions/inject-data.md (hunk 7 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:45]: Modified it to be a real-life example.

File: docs/Actions/inject-data.md (hunk 8 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:118]: Ok, I updated the example.

File: docs/Actions/inject-data.md (hunk 10 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:48]: Done it.

File: docs/Actions/inject-data.md (hunk 9 above)
Comment (Mihael03) [reply docs/Actions/inject-data.md:215]: Found one and added it

**Comment by bogdan-litescu** [comment]
Sorry, I put this on-hold until we figure out if we merge the 3 actions (inject data, execute tokens, apply tokens)

**Pull request merged**